http/objectstore/iam/bucket-metadata-sys.go

```diff
@@ -0,0 +1,169 @@
+func newBucketMetadataSys() *bucketMetadataSys {
+	return &bucketMetadataSys{
+		db: uleveldb.DBClient,
+	}
+}
+
+// Tags is list of tags of XML request/response as per
+// https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketTagging.html#API_GetBucketTagging_RequestBody
+type Tags tagging
+type tagging struct {
+	XMLName xml.Name `xml:"Tagging"`
+	TagSet  *TagSet  `xml:"TagSet"`
+}
+
+// TagSet represents list of unique tags.
+type TagSet struct {
+	TagMap   map[string]string
+	IsObject bool
+}
+
+// BucketMetadata contains bucket metadata.
+type BucketMetadata struct {
+	Name          string
+	Region        string
+	Created       time.Time
+	PolicyConfig  *policy.Policy
+	TaggingConfig *Tags
+}
+
+// newBucketMetadata creates BucketMetadata with the supplied name and Created to Now.
+func newBucketMetadata(name, region, accessKey string) BucketMetadata {
+	var p = policy.Policy{
+		ID:      policy.DefaultPolicies[0].Definition.ID,
+		Version: policy.DefaultPolicies[0].Definition.Version,
+		Statements: []policy.Statement{{
+			SID:       policy.DefaultPolicies[0].Definition.Statements[0].SID,
+			Effect:    policy.DefaultPolicies[0].Definition.Statements[0].Effect,
+			Principal: policy.NewPrincipal(accessKey),
+			Actions:   policy.DefaultPolicies[0].Definition.Statements[0].Actions,
+			Resources: policy.NewResourceSet(policy.NewResource(name, "*")),
+		}},
+	}
+	return BucketMetadata{
+		Name:         name,
+		Region:       region,
+		Created:      time.Now().UTC(),
+		PolicyConfig: &p,
+	}
+}
+
+// GetPolicyConfig returns configured bucket policy
+func (sys *bucketMetadataSys) GetPolicyConfig(bucket, accessKey string) (*policy.Policy, error) {
+	meta, err := sys.GetMeta(bucket, accessKey)
+	if err != nil {
+		if errors.Is(err, api_errors.ErrConfigNotFound) {
+			return nil, bucketPolicyNotFound{Bucket: bucket}
+		}
+		return nil, err
+	}
+	if meta.PolicyConfig == nil {
+		return nil, bucketPolicyNotFound{Bucket: bucket}
+	}
+	return meta.PolicyConfig, nil
+}
+
+// GetMeta returns a specific configuration from the bucket metadata.
+func (sys *bucketMetadataSys) GetMeta(bucket, accessKey string) (BucketMetadata, error) {
+	var meta BucketMetadata
+	err := sys.GetBucketMeta(bucket, accessKey, &meta)
+	if err != nil {
+		return BucketMetadata{}, err
+	}
+	return meta, nil
+}
+
+// SetBucketMeta - sets a new metadata in-db
+func (sys *bucketMetadataSys) SetBucketMeta(bucket, username string, meta BucketMetadata) error {
+	err := sys.db.Put(bucketPrefix+username+"-"+bucket, meta)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetBucketMeta metadata for a bucket.
+func (sys *bucketMetadataSys) GetBucketMeta(bucket, username string, meta *BucketMetadata) error {
+	err := sys.db.Get(bucketPrefix+username+"-"+bucket, meta)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// HeadBucketMeta metadata for a bucket.
+func (sys *bucketMetadataSys) HeadBucketMeta(bucket, username string) bool {
```

Comment:
If use this method to tell whether a bucket has in your system, why name it HasBucketMeta?

Reply:
It was used in "headbuckethandler" before, so it was named "headbucketmeta", now it is used in many functions, but I forgot to change the name, I will correct it later